flood-fill.py

```diff
@@ -1,14 +1,89 @@
+from pprint import pprint
 grid = [
         ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R'],
-        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R']
-        ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R']
+        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R'],
+        ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R'],
         ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R']
 ]
 
+
 def fill(row, col, new_color):
-    return # TODO practice for geoff
\ No newline at end of file
+        global grid
+        old_color = grid[row][col]
+        navigate(row, col, new_color, old_color)
+        return grid
+
+
+def navigate(row, col, new_color, old_color):
+        global grid
+        print("Checking grid[{}][{}]".format(row, col))
+        directions = {"north": ["row", -1], "south": ["row", 1], "east": ["col", 1], "west": ["col", -1]}
+        for direction in directions.values():
+                if grid[row][col] == old_color:
+                        grid[row][col] = new_color
+                        print("Just changed grid[{}][{}]".format(row, col))
+                if direction[0] == "row":
+                        try:
+                                if grid[row + direction[1]][col] != old_color:
+                                        continue
+                                else:
+                                        navigate(row + direction[1], col, new_color, old_color)
+                        except IndexError:
+                                continue
+                if direction[0] == "col":
+                        try:
+                                if grid[row][col + direction[1]] != old_color:
+                                        continue
+                                else:
+                                        navigate(row, col + direction[1], new_color, old_color)
+                        except IndexError:
+                                continue
+
+
+        return
```

Comment:
don't need to return if you return nothing, its the same as leaving it off

Reply:
makes sense